inc/dashboard/namespace.php

```diff
@@ -0,0 +1,456 @@
+		$conversion_rate = $block['conversionRate']['value'] ?? 0;
+
+		$data[ $block_id ]['views'] = $views;
+		$data[ $block_id ]['conversions'] = $conversions;
+		$data[ $block_id ]['avg_conversion_rate'] = $conversion_rate;
+	}
+
+	return $data;
+}
+
+/**
+ * Sort data by conversion rate.
+ *
+ * @param array $list The array of analytics data by block id.
+ * @param string $order The order to sort by. Accepted values are 'asc' or 'desc'.
+ * @param string $sort The parameter to sort by. Accepted values are 'views' and 'avg_conversion_rate'. Default is 'views'.
+ *
+ * @return array The sorted array.
+ */
+function sort_by( array $list, string $order = 'desc', string $sort = 'views' ) : array {
+	// If an invalid value was passed to $order, default to 'desc'.
+	if ( ! in_array( $order, [ 'asc', 'desc' ], true ) ) {
+		$order = 'desc';
+	}
+
+	// If an invalid value was passed to $sort, default to 'views'.
+	if ( ! in_array( $sort, [ 'avg_conversion_rate', 'views' ], true ) ) {
+		$sort = 'views';
+	}
+
+	$sort_order = ( $order === 'desc' ) ? SORT_DESC : SORT_ASC;
+	$sort = array_column( $list, $sort );
+
+	array_multisort( $sort, $sort_order, $list );
+
+	return $list;
+}
+
+/**
+ * Get analytics views list data.
+ *
+ * @param int $days How many days worth of analytics to fetch. Defaults to 7.
+ */
+function get_views_list( int $days = 7 ) : array {
+	$date_start = time() - ( $days * DAY_IN_SECONDS );
+
+	$query = [
+		'query' => [
+			'bool' => [
+				'filter' => [
+					[
+						// Query from the current site.
+						'term' => [
+							'attributes.blogId.keyword' => get_current_blog_id(),
+						],
+					],
+					[
+						// We're interested in views and conversions.
+						'terms' => [
+							'event_type.keyword' => [
+								'experienceView',
+								'conversion',
+							],
+						],
+					],
+					[
+						// Limit to date range.
+						'range' => [
+							'event_timestamp' => [
+								'gte' => $date_start * 1000,
+							],
+						],
+					],
+				],
+			],
+		],
+		'aggs' => [
+			'blocks' => [
+				'terms' => [
+					// Get the block data. This will give us the key for the block, which is stored as the post slug.
+					'field' => 'attributes.clientId.keyword',
+					'size' => 10000, // Use arbitrary large size that is more than we're likely to need.
+				],
+				'aggs' => [
+					'views' => [
+						'filter' => [
+							'term' => [
+								'event_type.keyword' => 'experienceView',
+							],
+						],
+						'aggs' => [
+							'uniques' => [
+								'cardinality' => [
+									'field' => 'endpoint.Id.keyword',
+								],
+							],
+						],
+					],
+					'conversions' => [
+						'filter' => [
+							'term' => [
+								'event_type.keyword' => 'conversion',
+							],
+						],
+						'aggs' => [
+							'uniques' => [
+								'cardinality' => [
+									'field' => 'endpoint.Id.keyword',
+								],
+							],
+						],
+					],
+					'conversionRate' => [
```

Comment:
If you prefer not to use camel case here you don't have to :)

Reply:
switched to snake case because that's what `bucket_script` uses